plugins/src/webrtcsink/imp.rs

```diff
@@ -787,81 +759,187 @@ impl CongestionController {
         }
     }
 
-    fn clamp_bitrate(&mut self, bitrate: i32, n_encoders: i32) {
-        self.target_bitrate = bitrate.clamp(
-            self.min_bitrate as i32 * n_encoders,
-            self.max_bitrate as i32 * n_encoders,
-        );
+    fn clamp_bitrate(&mut self, bitrate: i32, n_encoders: i32, controller_type: ControllerType) {
+        match controller_type {
+            ControllerType::Loss => {
+                self.target_bitrate_on_loss = bitrate.clamp(
+                    self.min_bitrate as i32 * n_encoders,
+                    self.max_bitrate as i32 * n_encoders,
+                )
+            }
+
+            ControllerType::Delay => {
+                self.target_bitrate_on_delay = bitrate.clamp(
+                    self.min_bitrate as i32 * n_encoders,
+                    self.max_bitrate as i32 * n_encoders,
+                )
+            }
+        }
+    }
+
+    fn get_remote_inbound_stats(&self, stats: &gst::StructureRef) -> Vec<gst::Structure> {
+        let mut inbound_rtp_stats: Vec<gst::Structure> = Default::default();
+        for (_, field_value) in stats {
+            if let Ok(s) = field_value.get::<gst::Structure>() {
+                if let Ok(type_) = s.get::<gst_webrtc::WebRTCStatsType>("type") {
+                    if type_ == gst_webrtc::WebRTCStatsType::RemoteInboundRtp {
+                        inbound_rtp_stats.push(s);
+                    }
+                }
+            }
+        }
+
+        inbound_rtp_stats
+    }
+
+    fn lookup_rtt(&self, stats: &gst::StructureRef) -> f64 {
+        let inbound_rtp_stats = self.get_remote_inbound_stats(stats);
+        let mut rtt = 0.;
+        let mut n_rtts = 0u64;
+        for inbound_stat in &inbound_rtp_stats {
+            if let Err(err) = (|| -> Result<(), gst::structure::GetError> {
+                rtt += inbound_stat.get::<f64>("round-trip-time")?;
+                n_rtts += 1;
+
+                Ok(())
+            })() {
+                gst_debug!(CAT, "{:?}", err);
+            }
+        }
+
+        rtt /= f64::max(1., n_rtts as f64);
+
+        gst_log!(CAT, "Round trip time: {}", rtt);
+
+        rtt
     }
 
-    fn control(
+    fn loss_control(
         &mut self,
         element: &super::WebRTCSink,
         stats: &gst::StructureRef,
         encoders: &mut Vec<VideoEncoder>,
     ) {
-        let n_encoders = encoders.len() as i32;
+        let loss_percentage = stats.get::<f64>("packet-loss-pct").unwrap();
 
-        let rtt = lookup_remote_inbound_rtp_stats(stats)
-            .and_then(|s| s.get::<f64>("round-trip-time").ok())
-            .unwrap_or(0.);
+        gst_error!(CAT, "Loss percentage: {}", loss_percentage);
```

Comment:
I know we're happy to have found the value, but maybe we don't need to proclaim it that loudly :)